### DISCUSSION TITLE
When downloading a file in Docker there's no extension

### DISCUSSION THREAD

**Discussion by motizukilucas** [cypress-io/cypress#27135] [Questions and Help]
Hi all, good morning/afternoon/evening wherever/whenever you are

I have a test in Cypress that click on a button and downloads a file

When running this test directly in my host (Ubuntu), the download is completed with _.docx_ extension

But when I try it inside Docker file is downloaded without extension

Is it supposed to happen like this?

I'm using `cypress/browsers:node12.14.1-chrome85-ff81` image which is the one recommended for CI setups

**Comment by portyu9** [comment]
Hi @motizukilucas 

Cypress itself should not be adding or removing the `.docx` extension. The **browser decides the downloaded filename**.

The first thing I would compare between the host run and Docker run is the download response. Ideally the server should send something explicit like:

```http
Content-Disposition: attachment; filename="report.docx"
Content-Type: application/vnd.openxmlformats-officedocument.wordprocessingml.document
```

If the application creates a browser download through an `<a>` element instead, give the `download` attribute the complete filename:

```html
<a href="..." download="report.docx">Download</a>
```

Without an explicit filename, browsers can infer one from the URL and MIME type, and that behavior can vary between browser versions.

That is particularly relevant here because:

```text
cypress/browsers:node12.14.1-chrome85-ff81
```

contains an extremely old Chrome/Node stack. I would reproduce this with a current Cypress Docker image and pin the **full immutable image tag** containing the exact Cypress/Node/browser versions you want, rather than continuing to diagnose Chrome 85 behavior.

Once the server/download naming is explicit, you can assert the result directly:

```js
cy.readFile('cypress/downloads/report.docx', null)
  .should('exist')
```

So I would investigate the response headers/download attribute and browser-version difference before adding any Cypress-side rename workaround.

The full discussion was reviewed; there were no screenshot attachments or replies.

Please mark this answer as accepted if it helped, thank you.
